back/internal/usersvc/internal/userdb/qryuser.go

```diff
@@ -62,19 +64,42 @@ func (s *UserDB) upsertUser(ctx cx, sid string, x *pb.AddUserReq, y *pb.UserResp
 		return err
 	}
 
-	r := pb.User{}
+	// Add role and user to user_role join table.
+	stmt, err := s.db.Prepare("INSERT into user_role (user_id, role_id) VALUES (?, ?)")
+	if err != nil {
+		return err
+	}
+
+	// Add entries to role table for every role
+	for _, rid := range x.RoleIds {
+		_, err = stmt.Exec(&id, rid)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
For now, just a thought experiment: How many queries might be called here? Is a transaction worthwhile? Can we reduce the query count?

Reply:
I'll create an issue.